app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -70,26 +79,120 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
         viewModel.loadDetails(txId)
     }
 
-    private fun updateUi(txDetails: TransferDetails?, isLoading: Boolean) {
+    private lateinit var contentBinding: ViewBinding
+
+    private fun updateUi(txDetails: TransactionDetails?, isLoading: Boolean) {
+
         binding.refresh.isRefreshing = isLoading
-        txDetails?.let {
-            binding.content.visible(true)
-            binding.txConfirmations.setExecutionData(
-                status = txDetails.txStatus,
-                confirmations = (txDetails.detailedExecutionInfo as? MultisigExecutionDetails)?.confirmations?.map { it.signer } ?: listOf(),
-                threshold = (txDetails.detailedExecutionInfo  as? MultisigExecutionDetails)?.confirmationsRequired ?: 0,
-                executor = txDetails.executor
-            )
-            binding.created.value = txDetails.createdAt?.formatBackendDate() ?: ""
-            binding.executed.value = txDetails.executedAt?.formatBackendDate() ?: ""
-            binding.etherscan.setOnClickListener {
-                requireContext().openUrl(
-                    getString(
-                        R.string.etherscan_transaction_url,
-                        txDetails.txHash
+
+        when (txDetails) {
+            is TransferDetails -> {
+                val viewStub = binding.stubTransfer
+                if (viewStub.parent != null) {
+                    val inflate = viewStub.inflate()
+                    contentBinding = TxDetailsTransferBinding.bind(inflate)
+                }
+                val txDetailsTransferBinding = contentBinding as TxDetailsTransferBinding
+
+                txDetailsTransferBinding.txConfirmations.setExecutionData(
+                    status = txDetails.txStatus,
+                    confirmations = (txDetails.detailedExecutionInfo as? MultisigExecutionDetails)?.confirmations?.map { it.signer } ?: listOf(),
+                    threshold = (txDetails.detailedExecutionInfo as? MultisigExecutionDetails)?.confirmationsRequired ?: 0,
+                    executor = txDetails.executor
+                )
+                txDetailsTransferBinding.created.value = txDetails.createdAt?.formatBackendDate() ?: ""
+
+                txDetailsTransferBinding.executed.value = txDetails.executedAt?.formatBackendDate() ?: ""
+                txDetailsTransferBinding.etherscan.setOnClickListener {
+                    requireContext().openUrl(
+                        getString(
+                            R.string.etherscan_transaction_url,
+                            txDetails.txHash
+                        )
                     )
+                }
+                val txType = if (txDetails.incoming == true) TxStatusView.TxType.TRANSFER_INCOMING else TxStatusView.TxType.TRANSFER_OUTGOING
+                txDetailsTransferBinding.txStatus.setStatus(txType, txDetails.txStatus)
+
+                if (txDetails.txData != null) {
```

Comment:
This view should not show the txData section (see https://zpl.io/2GJA7RJ)

Reply:
What about: https://zpl.io/bJmODxx (Incoming Transfer)

I think the Designs are not enough as a specification.

Reply:
Incoming custom transfers don't exist